fontbe/src/hvar.rs

```diff
@@ -0,0 +1,207 @@
+                            return None;
+                        }
+                        // Only 1 value per region for our input
+                        assert!(values.len() == 1, "{} values?!", values.len());
+                        Some((
+                            region.to_write_fonts_variation_region(axes),
+                            values[0].ot_round(),
+                        ))
+                    })
+                    .collect(),
+            )
+        }
+
+        let mut var_idxes = Vec::new();
+
+        // if we have a single model, we can try to build a VariationStore with implicit variation
+        // indices (a single ItemVariationData, outer index 0, inner index => gid).
+        let direct_store = if single_model {
+            let mut direct_builder = VariationStoreBuilder::new_with_implicit_indices();
+            for deltas in &all_glyph_width_deltas {
+                var_idxes.push(direct_builder.add_deltas(deltas.clone()));
+            }
+            // sanity check
+            assert_eq!(
+                mem::take(&mut var_idxes),
+                (0..glyph_order.len() as u32).collect::<Vec<_>>()
+            );
+            // we don't use the returned (identity) map in this case
+            Some(direct_builder.build().0)
+        } else {
+            None
+        };
+
+        // also build an indirect VariationStore with a DeltaSetIndexMap to map gid => varidx
+        let mut indirect_builder = VariationStoreBuilder::new();
+        for deltas in all_glyph_width_deltas {
+            var_idxes.push(indirect_builder.add_deltas(deltas));
+        }
+        let (indirect_store, varidx_map) = indirect_builder.build();
+
+        let varidx_map: DeltaSetIndexMap = var_idxes
+            .into_iter()
+            .map(|idx| {
+                varidx_map
+                    .get(idx)
+                    .unwrap_or_else(|| panic!("varidx_map missing index {}", idx))
+            })
+            .collect();
+
+        // prefer the most compact representation between the two stores
+        let use_direct = if direct_store.is_some() {
+            let direct_store_size =
+                table_size(direct_store.as_ref().unwrap(), "ItemVariationStore")?;
+            let indirect_store_size = table_size(&indirect_store, "ItemVariationStore")?;
+            let varidx_map_size = table_size(&varidx_map, "DeltaSetIndexMap")?;
+
+            direct_store_size <= indirect_store_size + varidx_map_size
+        } else {
+            false
+        };
+
+        let hvar = if use_direct {
+            Hvar::new(
+                MajorMinor::VERSION_1_0,
+                direct_store.unwrap(),
+                None,
+                None,
+                None,
+            )
+        } else {
+            Hvar::new(
+                MajorMinor::VERSION_1_0,
+                indirect_store,
+                Some(varidx_map),
+                None,
+                None,
+            )
```

Comment:
Seems verbose, suggested change in https://github.com/googlefonts/fontc/pull/524

Reply:
applied